pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1151,8 +1147,34 @@ func (lbaas *LbaasV2) ensureOctaviaListener(lbID string, oldListeners []listener
 		listenerChanged := false
 		updateOpts := listeners.UpdateOpts{}
 
-		if connLimit != listener.ConnLimit {
-			updateOpts.ConnLimit = &connLimit
+		if svcConf.connLimit != listener.ConnLimit {
+			updateOpts.ConnLimit = &svcConf.connLimit
+			listenerChanged = true
+		}
+		updateOpts.InsertHeaders = &listener.InsertHeaders
+		listenerKeepClientIP := listener.InsertHeaders[annotationXForwardedFor] == "true"
```

Comment:
What if `annotationXForwardedFor` key doesn't exist?

Reply:
Then `listenerKeepClientIP` is false, which should be the same as when then key is set to "false"

Reply:
I agree with @sbueringer: If `annotationXForwardedFor` does not exist, `listenerKeepClientIP` will be `false`.

Reply:
Yes, I was thinking about Python...